packages/redux-router5/index.d.ts

```diff
@@ -0,0 +1,120 @@
+declare module "redux-router5" {
+    import { Router, Plugin, State, Route } from 'router5';
+    import { Store, Middleware, Reducer, Dispatch, Action } from 'redux';
+
+    export function router5Middleware(router: Router): Middleware;
+
+    export interface RouterState {
+        route: State | null;
+        previousRoute: State | null;
+        transitionRoute: State | null;
+        transitionError: any | null;
+    }
+
+    export const router5Reducer: Reducer<RouterState>;
+
+    export function reduxPlugin<TState extends RouterState>(dispatch: Dispatch<TState>): Plugin;
+
+    export function routeNodeSelector<TState extends RouterState>(routeNode: string, reducerKey?: string): (state: TState) => Route;
```

Comment:
Here at the end should be `(state: {router: TState}) => RouterState` instead of `(state: TState) => Route;`. Because by default (or maybe even always) `router5Reducer` is meant to be used as such:

```
combineReducers({
  // ...
  router: router5Reducer,
  // ...
})
```

And we should get `RouterState` instead of `Route` from it, because `Route` is for routes description, not for any state. And it's not needed to import it with that change :-)

Reply:
I went with `export function routeNodeSelector<TState extends { router: RouterState }>(routeNode: string, reducerKey?: string): (state: TState) => Route;` because I want to allow different root state not different RouterState.